src/v/coproc/pacemaker.cc

```diff
@@ -112,7 +114,26 @@ std::vector<errc> pacemaker::add_source(
       id, _shared_res, std::move(ctxs));
     const auto [itr, success] = _scripts.emplace(id, std::move(script_ctx));
     vassert(success, "Double coproc insert detected");
-    (void)itr->second->start();
+    vlog(coproclog.debug, "Adding source with id: {}", id);
+    (void)ss::with_gate(_gate, [this, id, itr = itr] {
+        return itr->second->start().handle_exception_type(
```

Comment:
what happens if `_scripts` is rehashed before this continuation with a copy of `itr` is run and the iterator is invalidated?

Reply:
i think that node hash map has pointer stability, but not iterator stability.

Reply:
Yup nice catch again, can't do this. Performing another lookup within the gate